src/sentry/api/endpoints/organization_searches.py

```diff
@@ -82,18 +83,29 @@ def post(self, request: Request, organization) -> Response:
                     {"detail": "Query {} already exists".format(result["query"])}, status=400
                 )
 
-            saved_search = SavedSearch.objects.create(
-                organization=organization,
-                type=result["type"],
-                name=result["name"],
-                query=result["query"],
-                sort=result["sort"],
-            )
-            analytics.record(
-                "organization_saved_search.created",
-                search_type=SearchType(saved_search.type).name,
-                org_id=organization.id,
-                query=saved_search.query,
-            )
-            return Response(serialize(saved_search, request.user))
+            try:
+                saved_search = SavedSearch.objects.create(
+                    organization=organization,
+                    type=result["type"],
+                    name=result["name"],
+                    query=result["query"],
+                    sort=result["sort"],
+                )
+                analytics.record(
+                    "organization_saved_search.created",
+                    search_type=SearchType(saved_search.type).name,
+                    org_id=organization.id,
+                    query=saved_search.query,
+                )
+                return Response(serialize(saved_search, request.user))
+            except IntegrityError:
+                return Response(
+                    {
+                        "detail": "The combination ({}, {}, {}) for query {} already exists.".format(
+                            organization.id, result["name"], result["type"], result["query"]
+                        )
+                    },
+                    status=400,
```

Comment:
Not sure if 400 is appropriate response code to return here, but it aligns with the error code returned when query already exists.